lib/rom/sql/extensions/postgres/commands.rb

```diff
@@ -12,22 +12,26 @@ module Create
           def insert(tuples)
             tuples.map do |tuple|
               relation.dataset.returning(*relation.columns).insert(tuple)
-            end.flatten
+            end.flatten.map { |t| relation.output_schema[t] }
           end
 
           # Executes multi_insert statement and returns inserted tuples
           #
           # @api private
           def multi_insert(tuples)
-            relation.dataset.returning(*relation.columns).multi_insert(tuples)
+            relation.dataset.returning(*relation.columns).multi_insert(tuples).map { |t|
+              relation.output_schema[t]
+            }
```

Comment:
@flash-gordon `with_input_tuples` is just a convenient method that yields tuples even if it's *a tuple* (not an array with tuples). We could just rename it (and maybe even move to core, IIRC it's only in rom-sql but could be in core)